Core/Core/Pages/PageDetails/PageDetailsViewController.swift

```diff
@@ -18,101 +18,162 @@
+            let color = context.contextType == .course ? courses.first?.color : groups.first?.color
+        else { return }
+        updateNavBar(subtitle: name, color: color)
+    }
+
+    func update() {
+        guard let page = page else { return }
+        titleSubtitleView.title = page.title
+        navigationItem.rightBarButtonItem = canEdit ? optionsButton : nil
+        webView.loadHTMLString(page.body)
+    }
+
+    @objc func showOptions(_ sender: UIBarButtonItem) {
+        let alert = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
         alert.addAction(AlertAction(NSLocalizedString("Edit", bundle: .core, comment: ""), style: .default) { [weak self] _ in
-            guard let vc = self, let page = vc.presenter.page else {
-                return
-            }
+            guard let self = self, let page = self.page else { return }
             guard let url = page.htmlURL?.appendingPathComponent("edit") else { return }
-            self?.env.router.route(to: url, from: vc, options: .modal(.formSheet, embedInNav: true))
+            self.env.router.route(to: url, from: self, options: .modal(.formSheet, embedInNav: true))
         })
-
-        if presenter.canDelete() {
+        if canDelete {
             alert.addAction(AlertAction(NSLocalizedString("Delete", bundle: .core, comment: ""), style: .destructive) { [weak self] _ in
                 self?.showDeleteConfirmation()
             })
         }
         alert.addAction(AlertAction(NSLocalizedString("Cancel", bundle: .core, comment: ""), style: .cancel))
+        alert.popoverPresentationController?.barButtonItem = sender
         env.router.show(alert, from: self, options: .modal())
     }
 
     func showDeleteConfirmation() {
         let alert = UIAlertController(title: NSLocalizedString("Are you sure you want to delete this page?", bundle: .core, comment: ""), message: nil, preferredStyle: .alert)
         alert.addAction(AlertAction(NSLocalizedString("Cancel", bundle: .core, comment: ""), style: .cancel))
         alert.addAction(AlertAction(NSLocalizedString("OK", bundle: .core, comment: ""), style: .destructive) { [weak self] _ in
-            self?.presenter.deletePage()
+            self?.deletePage()
         })
         env.router.show(alert, from: self)
     }
 
-    func update() {
-        guard let page = presenter.page else {
-            return
-        }
-
-        titleSubtitleView.title = page.title
-        webView?.loadHTMLString(page.body)
+    func deletePage() {
+        guard let page = page else { return }
+        env.api.makeRequest(DeletePageRequest(context: context, url: pageURL)) { [weak self] (_, _, error) in performUIUpdate {
+            guard let self = self else { return }
+            if let error = error {
+                return self.showError(error)
+            }
+            self.env.database.viewContext.delete(page)
```

Comment:
You need to call `save()` after this delete. I deleted a page, quit the app, and the page was back at the pages list because it still exists in the persistent store.